napari/_qt/qt_viewer_buttons.py

```diff
@@ -76,67 +106,15 @@ def dropEvent(self, event):
             self.viewer.layers.remove_selected()
 
 
-class QtNewPointsButton(QPushButton):
-    def __init__(self, viewer):
-        super().__init__()
-
-        self.viewer = viewer
-        self.setToolTip('New points layer')
-        self.clicked.connect(lambda: self.viewer.add_points())
-
-
-class QtNewShapesButton(QPushButton):
-    def __init__(self, viewer):
-        super().__init__()
-
-        self.viewer = viewer
-        self.setToolTip('New shapes layer')
-        self.clicked.connect(lambda: self.viewer.add_shapes())
-
-
-class QtNewLabelsButton(QPushButton):
-    def __init__(self, viewer):
-        super().__init__()
-
-        self.viewer = viewer
-        self.setToolTip('New labels layer')
-        self.clicked.connect(lambda: self.viewer._new_labels())
-
-
-class QtResetViewButton(QPushButton):
-    def __init__(self, viewer):
-        super().__init__()
-
-        self.viewer = viewer
-        self.setToolTip('Reset view')
-        self.clicked.connect(lambda: self.viewer.reset_view())
-
-
-class QtRollDimsButton(QPushButton):
-    def __init__(self, viewer):
-        super().__init__()
-
-        self.viewer = viewer
-        self.setToolTip('Roll dimensions order for display')
-        self.clicked.connect(lambda: self.viewer.dims._roll())
-
-
-class QtTransposeDimsButton(QPushButton):
-    def __init__(self, viewer):
-        super().__init__()
-
-        self.viewer = viewer
-        self.setToolTip('Transpose displayed dimensions')
-        self.clicked.connect(lambda: self.viewer.dims._transpose())
-
-
-class QtConsoleButton(QPushButton):
-    def __init__(self, viewer):
+class QtViewerPushButton(QPushButton):
+    def __init__(self, viewer, tooltip, button_name, slot=None):
```

Comment:
similar question @jni about more keyword arg? maybe not for `viewer` or `button_name`, but for `tooltip=None`, could then default on `button_name` if None

Reply:
maybe reorder `button_name` and `tooltip` and add `*` after `button_name`